src/routes/v2/address.ts

```diff
@@ -82,40 +89,28 @@ function root(
 }
 
 // Query the Insight API for details on a single BCH address.
+// Returns a Promise.
 async function detailsFromInsight(
   thisAddress: string,
   currentPage: number = 0
 ) {
   try {
-    // Use the default (and max) page size of 1000
-    // https://github.com/bitpay/insight-api#notes-on-upgrading-from-v03
-    const pageSize = 1000
-
     const legacyAddr = BITBOX.Address.toLegacyAddress(thisAddress)
 
     let path = `${process.env.BITCOINCOM_BASEURL}addr/${legacyAddr}`
 
     // Set from and to params based on currentPage and pageSize
     // https://github.com/bitpay/insight-api/blob/master/README.md#notes-on-upgrading-from-v02
-    const from = currentPage * pageSize
-    const to = from + pageSize
+    const from = currentPage * PAGE_SIZE
+    const to = from + PAGE_SIZE
     path = `${path}?from=${from}&to=${to}`
 
     // Query the Insight server.
-    const response = await axios.get(path)
```

Comment:
This like was `await`ing the call to Insight, so the function was not returning a Promise, it was returning a resolved value.